### PR TITLE
Literature up to 2012

### PR DESCRIPTION
**Pull request by h3ssto**

### CODE REVIEW

File: MYabrv.bib [hunk 1]
```diff
@@ -309,7 +324,8 @@ @String{Piscataway
 	@String{Miami = "Miami, FL, USA"}
 	@String{Montreal = "Montreal, Quebec, Canada"}
 	@String{LosAlamitos = "Los Alamitos, CA, USA"}
-	@String{GI = "Gesellschaft f{\"u}r Informatik"}
+
+@String{GI = "Gesellschaft f{\"u}r Informatik e.V."}
```
Comment (tthuem) [inline MYabrv.bib:328]: Have never seen this before in a literature entry. Please revert.

File: literature.bib [hunk 2]
```diff
@@ -4047,6 +4158,40 @@ @article{MJ:TAOSD14
...
+	publisher = Springer,	
+	address = BerlinHeidelberg,
+	month=APR,
+	year=2014,
+}
+
+@article{BCHH:IJC14,
+	title={{Optimization Bounds from Binary Decision Diagrams}},
+	author={Bergman, David and Cire, Andre A and van Hoeve, Willem-Jan and Hooker, John N},
+	journal={INFORMS J.\ on Computing},
```
Comment (tthuem) [inline literature.bib:4187]: Why no abbreviation for this journal? Is in here several times.

File: literature.bib [hunk 3]
```diff
@@ -5391,14 +5536,64 @@ @article{TKB+:SCP14
...
+}
+
+@article{DLP:ENTCS13,
+	title={{Multi-Core BDD Operations for Symbolic Reachability}},
+	author={van Dijk, Tom and Laarman, Alfons and van De Pol, Jaco},
+	journal={Electron. Notes Theor. Comput. Sci.},
+	volume={296},
+	pages={127--143},
+	publisher = Elsevier,
+	year=2013
```
Comment (tthuem) [inline literature.bib:5556]: address missing

File: literature.bib [hunk 4]
```diff
@@ -5391,14 +5536,64 @@ @article{TKB+:SCP14
...
+}
+
+@book{ST13,
+	title={{The Satisfiability Problem: Algorithms and Analyses}},
+	author={Sch{\"o}ning, Uwe and Tor{\'a}n, Jacobo},
+	volume={3},
+	publisher={Lehmanns},
+	year=2013,
```
Comment (tthuem) [inline literature.bib:5564]: address missing

File: literature.bib [hunk 5]
```diff
@@ -5391,14 +5536,64 @@ @article{TKB+:SCP14
...
+
+@article{CH:JOR13,
+	title={{Multivalued Decision Diagrams for Sequencing Problems}},
+	author={Cir{\'e}, Andr{\'e} A and van Hoeve, Willem-Jan},
+	journal={J.\ on Operations Research},
+	volume={61},
+	number={6},
+	pages={1411--1428},
+	publisher=INFORMS,
+	year=2013
```
Comment (tthuem) [inline literature.bib:5575]: address missing

File: literature.bib [hunk 6]
```diff
@@ -6797,6 +6992,38 @@ @inproceedings{RAK+:VaMoS13
...
+@article{M:JSW12,
+	title={Biddy - a Multi-platform Academic {BDD} Package},
+	author={Meolic, Robert},
+	journal={J.\ of Software (JSW)},
+	volume={7},
+	number={6},
+	pages={1358--1366},
+	publisher = {{International Academy Publishing (IAP)}},
+	address = {San Bernardino, CA, USA},
+	year = 2012
```
Comment (tthuem) [inline literature.bib:7014]: Why no shorthand for journal, publisher, address?

File: literature.bib [hunk 7]
```diff
@@ -2198,6 +2198,97 @@ @inproceedings{FMS+:SANER17
...
+	journal   = JEA,
+	volume    = {21},
+	number    = {1},
+	month     = NOV,
+	pages     = {1.12:1--1.12:44},
+	doi       = {10.1145/2975585},
+	publisher = ACM,
+	address   = NY,
+	year      = 2016,
+}
```
Comment (tthuem) [inline literature.bib:2233]: It is not a big deal, but before I always tried to have spacing consistent:

"    key = value,"

**Review by tthuem — CHANGES_REQUESTED** [review]
Only minor comments. Lucky to see several problems fixed and some of my advised theses added. :)

**Review by tthuem — CHANGES_REQUESTED** [review]
Ignore this comment. Bug in Github made me write the comment twice...

File: MYabrv.bib (hunk 1 above)
Comment (h3ssto) [reply MYabrv.bib:328]: I have seen it in multiple entries and also it is the only "legal" spelling of the organisation, but will revert.

File: literature.bib (hunk 7 above)
Comment (h3ssto) [reply literature.bib:2233]: Normalized all spaces with `sed -e "s/\s*=\s*/ = /g"`

File: literature.bib (hunk 2 above)
Comment (h3ssto) [reply literature.bib:4187]: Added.

File: literature.bib (hunk 3 above)
Comment (h3ssto) [reply literature.bib:5556]: Fixed.

File: literature.bib (hunk 4 above)
Comment (h3ssto) [reply literature.bib:5564]: Fixed.

File: literature.bib (hunk 5 above)
Comment (h3ssto) [reply literature.bib:5575]: fixed.

File: literature.bib (hunk 6 above)
Comment (h3ssto) [reply literature.bib:7014]: First occurrence of journal, publisher, address.

File: literature.bib [hunk 8]
```diff
@@ -3263,6 +3354,7 @@ @inproceedings{TWS+:VaMoS16
 	year = 2016
 }
 
+% Warning: The phdthesis by P{\'e}rez is found P16:UNED
```
Comment (pmbittner) [inline literature.bib:3357]: good idea!

File: literature.bib [hunk 9]
```diff
@@ -3293,6 +3385,25 @@ @article{RTS+:JLAMP16
 
 %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%% 2015 %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
 
+@inproceedings{PyEDA15,
+	author 		= {Chris Drake},
+	title  		= {{PyEDA: Data Structures and Algorithms for Electronic Design Automation}},
+	booktitle 	= {{Proc.\ Conf.\ on Python in Science (SciPy)}},
```
Comment (pmbittner) [inline literature.bib]: Is there a reasony why this is not a variable in MyAbrv?

**Review by pmbittner — COMMENTED** [review]
I will have to do a second pass. Git said, my diff is outdated.

File: MYshort.bib [hunk 10]
```diff
@@ -314,12 +319,10 @@ @String{Piscataway
 	@String{Miami = "Miami, FL, USA"}
 	@String{Montreal = "Montreal, Quebec, Canada"}
 	@String{LosAlamitos = "Los Alamitos, CA, USA"}
-	@String{GI = "Gesellschaft f{\"u}r Informatik"}
-	@String{Bonn = "Bonn, Germany"}
-	@String{Singapore = "Singapore, Singapore"}
 
 @String{INFORMS = "INFORMS"}
  	@String{Catonsville = "Catonsville, USA"}
+ 	@String{Linthium = "Institute for Operations Research and the Management Sciences (INFORMS), Linthicum, MD, USA"}
```
Comment (pmbittner) [inline MYshort.bib:325]: Would it make sense to abbreviate this in MYshort as "INFORMS, Linthicum, MD, USA"?

File: literature.bib (hunk 9 above)
Comment (h3ssto) [reply literature.bib]: Single occurrence, doubtful there will be more.

File: literature.bib [hunk 11]
```diff
@@ -3386,22 +3386,22 @@ @article{RTS+:JLAMP16
...
 @inproceedings{PyEDA15,
-	author 		= {Chris Drake},
-	title  		= {{PyEDA: Data Structures and Algorithms for Electronic Design Automation}},
-	booktitle 	= {{Proc.\ Conf.\ on Python in Science (SciPy)}},
-  	pages     	= {25--30},
-	editor  	= {Kathryn Huff and James Bergstra},
-	year   		= 2015,
+	author = {Chris Drake},
+	title = {{PyEDA: Data Structures and Algorithms for Electronic Design Automation}},
+	booktitle = {{Proc.\ Conf.\ on Python in Science (SciPy)}},
```
Comment (pmbittner) [inline literature.bib:3391]: Is there a reason why this is not in MYabrv and MYshort?

File: MYshort.bib (hunk 10 above)
Comment (h3ssto) [reply MYshort.bib:325]: I copied the address from an entry of 2004, and just shortended Maryland -> MD, would happily drop the non-address part, too.

File: literature.bib [hunk 12]
```diff
@@ -4096,7 +4096,7 @@ @inproceedings{LG:VaMoS15
 @inproceedings{M:SE15,
 	author = {Mustafa Al-Hajjaji},
 	title = {{Scalable Sampling and Prioritization for Product-Line Testing}},
-	booktitle =  {Doktorandensymposium der Software Engineering},
+	booktitle = {Doktorandensymposium der Software Engineering},
```
Comment (pmbittner) [inline literature.bib:4099]: same here. Is this the complete name?

File: literature.bib [hunk 13]
```diff
@@ -4159,37 +4159,37 @@ @article{MJ:TAOSD14
...
 @article{BCHY:Heuristics14,
-	title={{BDD-Based Heuristics for Binary Optimization}},
-	author={Bergman, David and Cir{\'{e}}, Andr{\'{e}} A and van Hoeve, Willem-Jan and Yunes, Tallys},
-	journal={J.\ of Heuristics},
-	volume={20},
-	number={2},
-	pages={211--234},
+	title = {{BDD-Based Heuristics for Binary Optimization}},
+	author = {Bergman, David and Cir{\'{e}}, Andr{\'{e}} A and van Hoeve, Willem-Jan and Yunes, Tallys},
+	journal = {J.\ of Heuristics},
```
Comment (pmbittner) [inline literature.bib:4174]: same here

File: literature.bib [hunk 14]
```diff
@@ -5537,52 +5537,55 @@ @article{TKB+:SCP14
...
 
 @article{DLP:ENTCS13,
-	title={{Multi-Core BDD Operations for Symbolic Reachability}},
-	author={van Dijk, Tom and Laarman, Alfons and van De Pol, Jaco},
-	journal={Electron. Notes Theor. Comput. Sci.},
-	volume={296},
-	pages={127--143},
+	title = {{Multi-Core BDD Operations for Symbolic Reachability}},
+	author = {van Dijk, Tom and Laarman, Alfons and van De Pol, Jaco},
+	journal = {Electron. Notes Theor. Comput. Sci.},
```
Comment (pmbittner) [inline literature.bib:5552]: same here

File: literature.bib [hunk 15]
```diff
@@ -5537,52 +5537,55 @@ @article{TKB+:SCP14
...
-	author={Cir{\'e}, Andr{\'e} A and van Hoeve, Willem-Jan},
-	journal={J.\ on Operations Research},
-	volume={61},
-	number={6},
-	pages={1411--1428},
-	publisher=INFORMS,
-	year=2013
+	title = {{Multivalued Decision Diagrams for Sequencing Problems}},
+	author = {Cir{\'e}, Andr{\'e} A and van Hoeve, Willem-Jan},
+	journal = {J.\ on Operations Research},
```
Comment (pmbittner) [inline literature.bib:5572]: same here

File: literature.bib [hunk 16]
```diff
@@ -6993,35 +6996,35 @@ @inproceedings{RAK+:VaMoS13
...
 
 @inproceedings{LCF+:TASE12,
-	title={A Succinct and Efficient Implementation of a 2\^{} 32 BDD Package},
-	author={Lv, Guanfeng and Chen, Yao and Feng, Yachao and Chen, Qingliang and Su, Kaile},
-	booktitle={Proc.\ Int'l Symposium on Theoretical Aspects of Software Engineering (TASE)},
-	pages={241--244},
-	publisher=IEEE,
+	title = {A Succinct and Efficient Implementation of a 2\^{} 32 BDD Package},
+	author = {Lv, Guanfeng and Chen, Yao and Feng, Yachao and Chen, Qingliang and Su, Kaile},
+	booktitle = {Proc.\ Int'l Symposium on Theoretical Aspects of Software Engineering (TASE)},
```
Comment (pmbittner) [inline literature.bib:7001]: same here

File: literature.bib [hunk 17]
```diff
@@ -6993,35 +6996,35 @@ @inproceedings{RAK+:VaMoS13
...
 @article{M:JSW12,
-	title={Biddy - a Multi-platform Academic {BDD} Package},
-	author={Meolic, Robert},
-	journal={J.\ of Software (JSW)},
-	volume={7},
-	number={6},
-	pages={1358--1366},
+	title = {Biddy - a Multi-platform Academic {BDD} Package},
+	author = {Meolic, Robert},
+	journal = {J.\ of Software (JSW)},
```
Comment (pmbittner) [inline literature.bib:7011]: same here

File: literature.bib [hunk 18]
```diff
@@ -8681,25 +8684,25 @@ @inproceedings{KLG:MSR11
...
-  	pages={75--100},
- 	 publisher={Wiley Online Library},
- 	 year=2011
+ 	 author = {Cartaxo, Emanuela G and Machado, Patr{\'\i}cia DL and Neto, Francisco G Oliveira},
+ 	 title = {{On the Use of a Similarity Function for Test Case Selection in the Context of Model-Based Testing}},
+ 	 journal = STVR,
+  	volume = {21},
+  	number = {2},
+  	pages = {75--100},
+ 	 publisher = {Wiley Online Library},
```
Comment (pmbittner) [inline literature.bib:8693]: Isnt this already a variable in MYabrv?

File: literature.bib [hunk 19]
```diff
@@ -10691,15 +10694,15 @@ @article{MKB:JSME09
...
-	journal={Natural Computing},
-	volume={8},
-	number={2},
-	publisher={Springer Netherlands},
-	pages={239-287},
-	year=2009
+	author = {Bianchi, Leonora and Dorigo, Marco and Gambardella, LucaMaria and Gutjahr, WalterJ.},
+	title = {{A Survey on Metaheuristics for Stochastic Combinatorial Optimization}},
+	issn = {1567-7818},
+	journal = {Natural Computing},
```
Comment (pmbittner) [inline literature.bib:10700]: same here

File: literature.bib [hunk 20]
```diff
@@ -11999,13 +12002,13 @@ @inproceedings{JDB:SPLC07
...
-	author={Fraser, G. and Wotawa, F.}, 
-	title={{Mutant Minimization for Model-Checker Based Test-Case Generation}}, 
-	booktitle={in Proc. of Workshop on Mutation Analysis, Published with Proc. of Testing: Academic and Industrial Conference Practice and Research Techniques}, 
-	month=SEP, 
-	pages={161-168}, 
-	doi={10.1109/TAIC.PART.2007.30},
-	year=2007
+	author = {Fraser, G. and Wotawa, F.}, 
+	title = {{Mutant Minimization for Model-Checker Based Test-Case Generation}}, 
+	booktitle = {in Proc. of Workshop on Mutation Analysis, Published with Proc. of Testing: Academic and Industrial Conference Practice and Research Techniques},
```
Comment (pmbittner) [inline literature.bib:12007]: same here

File: literature.bib [hunk 21]
```diff
@@ -12106,11 +12109,11 @@ @inproceedings{LK+:ECBS07
...
 
 @article{NCB+SEI07,
-  	author={Northrop, Linda and Clements, P.},
-  	title={{A Framework for Software Product Line Practice, Version 5.0}},
- 	 journal={SEI},
-	address =  {Pittsburgh,USA},
-  	year=2007
+  	author = {Northrop, Linda and Clements, P.},
+  	title = {{A Framework for Software Product Line Practice, Version 5.0}},
+ 	 journal = {SEI},
```
Comment (pmbittner) [inline literature.bib:12114]: brackets too much?

**Review by pmbittner — CHANGES_REQUESTED** [review]
Hi Tobi,

allmost all my comments are the same, namely why some booktitles and journal values are not realized as a variable in MYabrv and MYshort. Because your fixes to spacing (which are appreciated) touched also lots of existing references, I could not backtrack whether my comments were on issues of your new/updated references from your pull-request or existing references. As it shouldn't be your task now to also fix existing references (or at least not in this PR), you might discard some of my comments if these were made on already existing references.

File: literature.bib (hunk 11 above)
Comment (h3ssto) [reply literature.bib:3391]: As before, single occurrence, will move to MYabrv/MYshort once a second occurrence is added.

File: literature.bib (hunk 12 above)
Comment (h3ssto) [reply literature.bib:4099]: I don't know, I just removed the additional space.

File: literature.bib (hunk 13 above)
Comment (h3ssto) [reply literature.bib:4174]: Single Occurence.

File: literature.bib (hunk 14 above)
Comment (h3ssto) [reply literature.bib:5552]: Single occurrence.

File: literature.bib (hunk 15 above)
Comment (h3ssto) [reply literature.bib:5572]: Single Occurrence.

File: literature.bib (hunk 16 above)
Comment (h3ssto) [reply literature.bib:7001]: Single Occurrence.

File: literature.bib (hunk 17 above)
Comment (h3ssto) [reply literature.bib:7011]: Single Occurrence.

File: literature.bib (hunk 18 above)
Comment (h3ssto) [reply literature.bib:8693]: This is neither of the Wiley organs present in MYabrv.

File: literature.bib (hunk 19 above)
Comment (h3ssto) [reply literature.bib:10700]: Not mine, just fixed the spaces.

File: literature.bib (hunk 20 above)
Comment (h3ssto) [reply literature.bib:12007]: Again, not mine, just fixed spaces.

File: literature.bib (hunk 21 above)
Comment (h3ssto) [reply literature.bib:12114]: Again, not mine, just fixed spaces, but will fix it, too.

Comment (h3ssto) [reply literature.bib]: This entry is originally from @tthuem, won't touch.

File: literature.bib (hunk 20 above)
Comment (h3ssto) [reply literature.bib:12007]: https://github.com/tthuem/BibTags/blame/master/literature.bib#L11776

File: literature.bib (hunk 19 above)
Comment (h3ssto) [reply literature.bib:10700]: https://github.com/tthuem/BibTags/blame/master/literature.bib#L10468

File: literature.bib (hunk 12 above)
Comment (h3ssto) [reply literature.bib:4099]: https://github.com/tthuem/BibTags/blame/master/literature.bib#L3985

**Comment by tthuem** [comment]
Great. Go ahead with the merge.

**Pull request merged**